File: tests/fixtures/dart/sample.dart
/// User service for managing users.
class UserService {
  /// Get user by ID.
  String getUser(int userId) {
    return 'user-$userId';
  }

  /// Delete a user.
  bool deleteUser(int userId) {
    return true;
  }
}

/// Authenticate a token.
bool authenticate(String token) {
  return token.isNotEmpty;
}

/// Status of a request.
enum Status { pending, active, done }

/// JSON map alias.
typedef JsonMap = Map<String, dynamic>;

File: tests/fixtures/typescript/sample.ts
const MAX_TIMEOUT: number = 5000;

interface User {
    id: number;
    name: string;
}

class UserService {
    getUser(userId: number): User {
        return { id: userId, name: "" };
    }
}

function authenticate(token: string): boolean {
    return token.length > 0;
}

type UserID = number;

export const createUser = mutation({
    args: { name: v.string() },
    handler: async (ctx, args) => { return ctx.db.insert('users', args); }
});

export const getUser = query({
    args: { id: v.id('users') },
    handler: async (ctx, args) => { return ctx.db.get(args.id); }
});

const API_BASE_URL = "https://api.example.com";
